From dcb087567b69453c6d9aa9378fa2dfb4c90e73cd Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Wed, 2 Dec 2020 14:19:18 +0700
Subject: [PATCH 324/355] arm64: dts: renesas: r8a774a3: Add support VSP
 instances

The r8a774a3 has 5 VSP instances.
This patch adds vspb and vspi for VSP Manager, and vspd(0,1,2) for
 vsp2 driver.
Also remove fcp nodes support for vspb and vspi device nodes because
they are already supported in VSP Manager nodes.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774a3.dtsi | 95 +++++++++++++++++++++++++------
 1 file changed, 78 insertions(+), 17 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a774a3.dtsi b/arch/arm64/boot/dts/renesas/r8a774a3.dtsi
index 9ac164c..aebec97 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a3.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a774a3.dtsi
@@ -2217,11 +2217,6 @@
 			renesas,#ch = <0>;
 		};
 
-		fcpvb0: fcp@fe96f000 {
-			reg = <0 0xfe96f000 0 0x200>;
-			/* placeholder */
-		};
-
 		fcpvd0: fcp@fea27000 {
 			reg = <0 0xfea27000 0 0x200>;
 			/* placeholder */
@@ -2237,44 +2232,110 @@
 			/* placeholder */
 		};
 
-		fcpvi0: fcp@fe9af000 {
-			reg = <0 0xfe9af000 0 0x200>;
-			/* placeholder */
-		};
-
 		vsp@fe960000 {
+			compatible = "renesas,vspm-vsp2";
 			reg = <0 0xfe960000 0 0x8000>;
-			/* placeholder */
+			renesas,has-bru;
+			renesas,has-lut;
+			renesas,has-clu;
+			renesas,has-hgo;
+			renesas,#rpf = <5>;
+			renesas,#uds = <0>;
+			renesas,#wpf = <1>;
 		};
 
 		vspb: vspm@fe960000 {
+			compatible = "renesas,vspm";
 			reg = <0 0xfe960000 0 0x8000>, <0 0xfe96f000 0 0x200>;
-			/* placeholder */
+			interrupts = <GIC_SPI 266 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 626>, <&cpg CPG_MOD 607>;
+			clock-names = "vsp", "fcp";
+			renesas,#ch = <3>;
+			renesas,#rpf = <31>;
+			renesas,#rpf_clut = <6>;
+			renesas,#wpf_rot = <0>;
+			renesas,has-lut;
+			renesas,has-clu;
+			renesas,has-bru;
+			renesas,has-hgo;
+			renesas,#read_outstanding = <0>;
+			renesas,#start_reservation = <1>;
+			power-domains = <&sysc R8A774A3_PD_A3VC>;
+			resets = <&cpg 626>, <&cpg 607>;
+			reset-names = "vsp", "fcp";
+			status = "disabled";
 		};
 
 		vspd0: vsp@fea20000 {
+			compatible = "renesas,vsp2";
 			reg = <0 0xfea20000 0 0x5000>;
-			/* placeholder */
+			interrupts = <GIC_SPI 466 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 623>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 623>;
+
+			renesas,fcp = <&fcpvd0>;
 		};
 
 		vspd1: vsp@fea28000 {
+			compatible = "renesas,vsp2";
 			reg = <0 0xfea28000 0 0x5000>;
-			/* placeholder */
+			interrupts = <GIC_SPI 467 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 622>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 622>;
+
+			renesas,fcp = <&fcpvd1>;
 		};
 
 		vspd2: vsp@fea30000 {
+			compatible = "renesas,vsp2";
 			reg = <0 0xfea30000 0 0x5000>;
-			/* placeholder */
+			interrupts = <GIC_SPI 468 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 621>;
+			power-domains = <&sysc R8A774A3_PD_ALWAYS_ON>;
+			resets = <&cpg 621>;
+
+			renesas,fcp = <&fcpvd2>;
 		};
 
 		vsp@fe9a0000 {
+			compatible = "renesas,vspm-vsp2";
 			reg = <0 0xfe9a0000 0 0x8000>;
-			/* placeholder */
+			renesas,has-lut;
+			renesas,has-clu;
+			renesas,has-hgo;
+			renesas,has-hgt;
+			renesas,#rpf = <1>;
+			renesas,#uds = <1>;
+			renesas,#wpf = <1>;
 		};
 
 		vspi0: vspm@fe9a0000 {
+			compatible = "renesas,vspm";
 			reg = <0 0xfe9a0000 0 0x8000>, <0 0xfe9af000 0 0x200>;
-			/* placeholder */
+			interrupts = <GIC_SPI 444 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 631>, <&cpg CPG_MOD 611>;
+			clock-names = "vsp", "fcp";
+			renesas,#ch = <0>;
+			renesas,#rpf = <1>;
+			renesas,#rpf_clut = <1>;
+			renesas,#wpf_rot = <1>;
+			renesas,has-sru;
+			renesas,has-uds;
+			renesas,has-lut;
+			renesas,has-clu;
+			renesas,has-hst;
+			renesas,has-hsi;
+			renesas,has-hgo;
+			renesas,has-hgt;
+			renesas,has-shp;
+			renesas,#read_outstanding = <2>;
+			renesas,#start_reservation = <1>;
+			power-domains = <&sysc R8A774A3_PD_A3VC>;
+			resets = <&cpg 631>, <&cpg 611>;
+			reset-names = "vsp", "fcp";
+			status = "disabled";
 		};
 
 		hdmi0: hdmi@fead0000 {
-- 
2.7.4

